openvpn/process_factory.go

```diff
@@ -26,10 +26,10 @@ import (
 )
 
 // CreateNewProcess creates new openvpn process with given config params
-func CreateNewProcess(openvpnBinary string, config *config.GenericConfig, middlewares ...management.Middleware) *OpenvpnProcess {
+func CreateNewProcess(openvpnBinary string, config *config.GenericConfig, lastSessionShutdown chan bool, middlewares ...management.Middleware) *OpenvpnProcess {
```

Comment:
Do we really need to pollute constructor with private attributes like `lastSessionShutdown`.
Does caller want to get "shutdown start" like event? Can it be taken by listening `Wait()`